File: modulesCore/login/src/main/java/com/example/modulescore/login/di/LoginActivityModule.java
package com.example.modulescore.login.di;

import com.example.modulescore.login.net.LoginService;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ActivityComponent;
import retrofit2.Retrofit;

@Module
@InstallIn(ActivityComponent.class)
public class LoginActivityModule {

    @Provides
    public LoginService provideLoginService(Retrofit retrofit) {
        return retrofit.create(LoginService.class);
    }


}

File: modulesCore/login/src/main/java/com/xupt/safeAndRun/modulesCore/login/di/LoginActivityModule.kt
package com.xupt.safeAndRun.modulesCore.login.di

import com.xupt.safeAndRun.modulesCore.login.net.ForgetService
import com.xupt.safeAndRun.modulesCore.login.net.LoginService
import com.xupt.safeAndRun.modulesCore.login.net.RegisterService
import com.xupt.safeAndRun.modulesCore.login.net.VerifyService
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ViewModelComponent
import retrofit2.Retrofit

@Module
@InstallIn(ViewModelComponent::class)
object LoginActivityModule {
    @Provides
    fun provideLoginService(retrofit: Retrofit): LoginService {
        return retrofit.create(LoginService::class.java)
    }
    @Provides
    fun provideRegisterService(retrofit: Retrofit): RegisterService {
        return retrofit.create(RegisterService::class.java)
    }
    @Provides
    fun provideVerifyService(retrofit: Retrofit): VerifyService {
        return retrofit.create(VerifyService::class.java)
    }
    @Provides
    fun provideForgetService(retrofit: Retrofit): ForgetService {
        return retrofit.create(ForgetService::class.java)
    }
}